runtime/doc/channels.txt

```diff
@@ -0,0 +1,91 @@
+*channel.txt*    Nvim
+
+
+		 NVIM REFERENCE MANUAL    by Thiago de Arruda
+
+
+Nvim's facilities for async io				  *channels*
+
+				      Type <M-]> to see the table of contents.
+
+==============================================================================
+1. Introduction						    *channels-intro*
+
+Channels are nvim's way of communicating with external processes.
+
+There are several ways to open a channel:
+
+  1. Through stdin/stdout when `nvim` is started with `--headless`, and a startup
+     script or --cmd  commad opens the stdio channel using |stdioopen()|.
+
+  2. Through stdin, stdout and stderr of a process spawned by |jobstart()|.
+
+  3. Through the PTY master end of a PTY opened with
+     `jobstart(..., {'pty': v:true})` or |termopen()|.
```

Comment:
I wouldn't mind it. But maybe in a later PR? After all, this PR does not introduce any new types.